interfaces/builtin/maliit_input_method.go

```diff
@@ -0,0 +1,152 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+var maliitInputMethodPermanentSlotAppArmor = []byte(`
+# Description: Allow operating as maliit server.
+
+# DBus accesses
+#include <abstractions/dbus-session>
+
+# allow binding to the maliit dbus server for address negotiation
+dbus (bind)
+    bus=session
+    name="org.maliit.server",
+
+/usr/share/glib-2.0/schemas/ r,
+
+# maliit uses peer-to-peer dbus over a unix socket after address negotiation
+unix (bind, send, receive, listen, accept) addr="@/tmp/maliit-server/dbus-*",
+`)
+
+var maliitInputMethodConnectedSlotAppArmor = []byte(`
+# Provide access to the maliit address service
+dbus (send, receive)
+    bus=session
+    interface="org.maliit.Server.Address"
+    path=/org/maliit/server/address
+    peer=(label=###PLUG_SECURITY_TAGS###),
+`)
+
+var maliitInputMethodConnectedPlugAppArmor = []byte(`
+# Description: Allow applications to connect to a maliit socket
+# Usage: common
+
+#include <abstractions/dbus-session>
+
+# Find the maliit input method socket
+dbus (send, receive)
+    bus=session
+    interface="org.maliit.Server.Address"
+    path=/org/maliit/server/address
+    peer=(label=###SLOT_SECURITY_TAGS###),
+
+unix (send, receive, connect) addr=none peer=(label=###SLOT_SECURITY_TAGS###),
```

Comment:
There's only actually two IPC mechanisms, 1 was unnecessary and has now been removed, 4 was the other side of the /tmp/maliit-server/dbus-* unix socket, app armor identifies this as having an addr=none but a peer_addr of /tmp/maliit-server/dbus-*, so I've updated the rule to include the peer_addr and have added further comments to clarify the remaining two IPC systems (the dbus address server and the unix socket).